src/tty.rs

```diff
@@ -202,17 +203,66 @@ fn get_pw_entry(buf: &mut [i8; 1024]) -> Passwd {
     }
 }
 
+fn parse_shell(raw: Option<&String>) -> Option<Vec<CString>> {
```

Comment:
I think this function can almost entirely disappear if we force `-e` to be the last argument.